File: tests/fixtures/flutter/lib/session.dart
import 'dart:io';
import 'package:shared_preferences/shared_preferences.dart';

class Session {
  Future<void> enregistrer(String jeton, String rafraichissement) async {
    final prefs = await SharedPreferences.getInstance();
    // Un jeton d'acces dans SharedPreferences : lisible en clair.
    await prefs.setString('auth_token', jeton);
    await prefs.setString('refresh_token', rafraichissement);

    // Correct : une preference d'affichage n'a rien de sensible.
    await prefs.setString('theme', 'sombre');
    await prefs.setInt('dernier_onglet', 2);
  }

  HttpClient client() {
    final client = HttpClient();
    client.badCertificateCallback = (cert, host, port) => true;
    return client;
  }
}

File: tests/fixtures/android-kotlin/app/src/main/java/com/exemple/Session.kt
package com.exemple

import android.content.Context
import android.content.SharedPreferences

class Session(context: Context) {
    private val prefs: SharedPreferences =
        context.getSharedPreferences("session", Context.MODE_PRIVATE)

    fun enregistrer(jeton: String, rafraichissement: String) {
        prefs.edit()
            .putString("auth_token", jeton)
            .putString("refresh_token", rafraichissement)
            .apply()
    }

    // Correct : une preference d'affichage n'a rien de sensible.
    fun choisirTheme(theme: String) {
        prefs.edit().putString("theme", theme).apply()
        prefs.edit().putInt("dernier_onglet", 2).apply()
    }
}
